data/lua/wml/harm_unit.lua

```diff
@@ -160,19 +160,52 @@ function wml_actions.harm_unit(cfg)
 
 			wesnoth.interface.float_label( unit_to_harm.x, unit_to_harm.y, string.format( "<span foreground='red'>%s</span>", text ) )
 
+			local xp_mode = {kill=false, attack=false, defend=false}
+			local experience_split = tostring(experience):split()
+			for _,opt in ipairs(experience_split) do
+				if opt == 'true' or opt == 'yes' or opt == 'nil' then
+					xp_mode.kill = true
+					xp_mode.attack = true
+					xp_mode.defend = true
+				elseif opt == 'fight' then
+					xp_mode.attack = true
+					xp_mode.defend = true
+				elseif opt == 'attack' then
+					xp_mode.attack = true
+				elseif opt == 'defend' then
+					xp_mode.defend = true
+				elseif opt == 'kill' then
+					xp_mode.kill = true
+				elseif not (opt == 'no' or opt == 'false') then
+					wml.error('Invalid [harm_unit] experience, list of values can contain: yes, kill, fight, attack, defend, no')
+				end
+				-- with 'no' or 'false' preserve previous value
+			end
+			if #experience_split == 0 then
+				xp_mode = {kill=true, attack=true, defend=true}
+			end
+
 			local function calc_xp( level ) -- to calculate the experience in case of kill
 				if level == 0 then return math.ceil(wesnoth.game_config.kill_experience / 2)
 				else return level * wesnoth.game_config.kill_experience end
 			end
 
-			if experience ~= false and harmer and harmer.valid
+			if harmer and harmer.valid
 				and wesnoth.sides.is_enemy( unit_to_harm.side, harmer.side )
 			then
-				if kill ~= false and unit_to_harm.hitpoints <= 0 then
-					harmer.experience = harmer.experience + calc_xp( unit_to_harm.level )
+				if unit_to_harm.hitpoints <= 0 then
+					if xp_mode.kill then
+						harmer.experience = harmer.experience + calc_xp( unit_to_harm.level )
+					elseif xp_mode.attack then
+						harmer.experience = harmer.experience + wesnoth.game_config.combat_experience * unit_to_harm.level
```

Comment:
I thought the eventual decision was that killing a unit gave no xp if `xp_mode.kill` was false.

Reply:
You convinced me to keep current behaviour.

Reply:
Ah, :grin: